src/util/rampingvalue.h

```diff
@@ -1,22 +1,58 @@
 #ifndef MIXXX_UTIL_RAMPINGVALUE_H
 #define MIXXX_UTIL_RAMPINGVALUE_H
 
+// RampingValue is a utility class for numeric values that need to be smoothly
+// ramped across each frame or sample of an audio buffer. It compares the
+// current value of a parameter at the time of an audio engine callback to that
+// parameter's value at the last cycle of the audio engine and smoothly
+// interpolates the intermediate values across the span of the buffer.
 
+// To use a RampingValue, store it as a variable that persists across audio
+// engine callbacks. The constructor takes the number of steps that will be
+// used when iterating over the audio buffer, which is typically the number of
+// frames or samples in the audio engine's buffer. At the start of each audio
+// engine cycle, before the parameter is used, call
+// RampingValue::setCurrentCallbackValue with the parameter value received from
+// outside the audio engine thread. Then, when looping over the audio buffer,
+// get the interpolated value with RampingValue::rampedValue
 template <typename T>
 class RampingValue {
   public:
-    RampingValue(const T& initial, const T& final, int steps) {
```

Comment:
Sorry to say, but my old implementation was way better, because it was designed to live on the fast stack.
Originally it requires only the old value to be in the state class, now we have four values that have to be fetched from the heap each call. The original getNext function only did a =+ return on the fast stack. Now we have four lines accessing heap memory. So please revert the changes.

Reply:
Okay, it will be easier to fix the bugs with Echo and Reverb starting in a new branch, so I'll close this PR.